src/parser/ad_parser.js

```diff
@@ -237,47 +236,70 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
   extensions.forEach(extNode => {
-    const ext = new AdExtension();
-    const extNodeAttrs = extNode.attributes;
-    const childNodes = extNode.childNodes;
+    const ext = parseExtension(extNode);
 
-    if (extNode.attributes) {
-      for (const extNodeAttrKey in extNodeAttrs) {
+    if (ext) {
+      collection.push(ext);
+    }
+  });
+}
+
+/**
+ * Parses an extension child node
+ * @param {Object} extNode - The extension node to parse
+ * @return {Object}
+ */
+function parseExtension(extNode) {
+  // Ignore comments
+  if (extNode.nodeName === '#comment') return null;
+
+  const ext = new AdExtension();
+  const extNodeAttrs = extNode.attributes;
+  const childNodes = extNode.childNodes;
+  const txt = parserUtils.parseNodeText(extNode);
+
+  if (txt !== '') {
+    ext.value = txt;
+    ext.name = extNode.nodeName;
+  }
+
+  // Parse attributes
+  if (extNode.attributes) {
+    for (const extNodeAttrKey in extNodeAttrs) {
+      if (extNodeAttrs.hasOwnProperty(extNodeAttrKey)) {
```

Comment:
is this check necessary? I mean for...in wouldnt output an `extNodeAttrKey` if it wasnt a property of `extNodeAttrs`, but maybe I'm missing something